functional/include/fun/option/option_inner.h

```diff
@@ -29,17 +29,18 @@ struct OptionUnion<T, std::enable_if_t<std::is_empty<T>::value>>: private T {
 
   Self& operator=(Self&& other) noexcept {
     if (this != &other) {
-      static_cast<T&>(*this) = static_cast<T&&>(other);
       _variant = other._variant;
+      if (_variant) { static_cast<T&>(*this) = static_cast<T&&>(other); }
       other._variant = Tag::NONE;
     }
+    return *this;
   }
 
   Self clone() const { return *this; }
 
-  OptionUnion() : _variant(Tag::NONE) {}
+  OptionUnion() : T(), _variant(Tag::NONE) {}
 
-  explicit OptionUnion(T val) : T(std::move(val)), _variant(Tag::SOME) {}
+  explicit OptionUnion(T arg) : T(std::move(arg)), _variant(Tag::SOME) {}
```

Comment:
Why the variable name change?

Reply:
That was accidental. I think I deleted that line at some point when I was trying a different route (that ended up being a dead end), and then re-wrote it, and incidentally my rewrite used a different variable name. This hunk should probably be discarded.

Reply:
Got it, can you please discard this hunk?

Reply:
Discarded